server/src/main/java/io/crate/metadata/table/Operation.java

```diff
@@ -81,15 +86,33 @@ public enum Operation {
             .put(IndexMetadata.SETTING_BLOCKS_METADATA, METADATA_DISABLED_OPERATIONS)
             .map();
 
-    public static EnumSet<Operation> buildFromIndexSettingsAndState(Settings settings, IndexMetadata.State state) {
+    public static EnumSet<Operation> buildFromIndexSettingsAndState(Settings settings, IndexMetadata.State state, RelationName ident) {
         if (state == IndexMetadata.State.CLOSE) {
             return CLOSED_OPERATIONS;
         }
         Set<Operation> operations = ALL;
+
         var subscriptionName = REPLICATION_SUBSCRIPTION_NAME.get(settings);
-        if (subscriptionName != null && subscriptionName.isEmpty() == false) {
+        var publicationNames = REPLICATION_PUBLICATION_NAMES.get(settings);
+        var isPublished = publicationNames != null && publicationNames.isEmpty() == false;
+        var isSubscribed = subscriptionName != null && subscriptionName.isEmpty() == false;
+
+        if (isPublished && isSubscribed) {
+            throw new IllegalStateException(
+                String.format(
+                    Locale.ENGLISH,
+                    "The table '%s' cannot be published and subscribed at the same time in the same cluster",
```

Comment:
I think we allow chaining, discussion [here](https://github.com/crate/crate/pull/11821#discussion_r729675412)

A(t1) -> B(t1) -> C(t1)
table t1 of the cluster B is such example - it's taken from subscription to C and published further to A

Reply:
Great feedback, I was not aware of that. Then I think the subscribed operation set should be applied, as it is more restrictive.